ddtrace/contrib/internal/aiokafka/patch.py

```diff
@@ -0,0 +1,201 @@
+        args, kwargs = set_argument_value(args, kwargs, 5, "headers", headers, override_unset=True)
+
+        try:
+            result = await func(*args, **kwargs)
+        except BaseException as e:
+            span.set_exc_info(type(e), e, e.__traceback__)
+            span.finish()
+            raise e
+
+        def sent_callback(future):
+            try:
+                result = future.result()
+                core.dispatch("aiokafka.send.completed", (result,))
+            except Exception as e:
+                span.set_exc_info(type(e), e, e.__traceback__)
+            finally:
+                span.finish()
+
+        result.add_done_callback(sent_callback)
+        return result
+
+
+async def traced_getone(func, instance, args, kwargs):
+    # we must get start time now since execute before starting a span in order to get distributed context
+    # if it exists
+    start_ns = time_ns()
+    err = None
+    message = None
+    parent_ctx = None
+
+    group_id = instance._group_id
+    bootstrap_servers = instance._client._bootstrap_servers
+
+    try:
+        message = await func(*args, **kwargs)
+        if config.aiokafka.distributed_tracing_enabled and message.headers:
+            dd_headers = {
+                key: (val.decode("utf-8", errors="ignore") if isinstance(val, (bytes, bytearray)) else str(val))
+                for key, val in message.headers
+                if val is not None
+            }
+            parent_ctx = HTTPPropagator.extract(dd_headers)
+    except Exception as e:
+        err = e
+        raise err
+    finally:
+        with core.context_with_data(
+            "aiokafka.getone",
+            call_trace=False,
+            span_name=schematize_messaging_operation(CONSUME, provider="kafka", direction=SpanDirection.INBOUND),
+            span_type=SpanTypes.WORKER,
+            service=trace_utils.ext_service(None, config.aiokafka),
+            distributed_context=parent_ctx,
+            tags=common_consume_aiokafka_tags(getattr(message, "topic", None), bootstrap_servers, group_id),
+        ) as ctx:
+            span = ctx.span
+            span.start_ns = start_ns
+
+            core.dispatch("aiokafka.getone.message", (instance, span, message, err))
+    return message
+
+
+async def traced_getmany(func, instance, args, kwargs):
+    group_id = instance._group_id
+    bootstrap_servers = instance._client._bootstrap_servers
+
+    with core.context_with_data(
+        "aiokafka.getmany",
+        call_trace=False,
+        span_name=schematize_messaging_operation(CONSUME, provider="kafka", direction=SpanDirection.INBOUND),
+        span_type=SpanTypes.WORKER,
+        service=trace_utils.ext_service(None, config.aiokafka),
+        tags=common_consume_aiokafka_tags(None, bootstrap_servers, group_id),
+    ) as ctx:
+        span = ctx.span
+        messages = await func(*args, **kwargs)
+
+        core.dispatch("aiokafka.getmany.message", (instance, span, messages))
```

Comment:
Passing the context rather than the span here would make this code slightly more generic